### PR TITLE
Improve image browsing and interface usability + automatic GCP detection

### PR DESCRIPTION
**Pull request by HeDo88TH**
This PR adds the following features:

- EXIF tag reading: first show the images that are closest to the GCP
- Image retention: no need to re-upload the images for each GCP
- Automatic GCP detection: opencv haar classifier to find GCPs in images

### CODE REVIEW

**Comment by pierotofy** [comment]
Closes #10

**Comment by HeDo88TH** [comment]
This is ready 🚀 @pierotofy

File: src/app/images-tagger/images-tagger.component.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
             <input type="file" class="custom-file-input" #imagesUpload (change)="imagesSelected($event)" accept="image/*"
                 multiple="true" id="imagesFiles">
             <label class="custom-file-label mb-0" for="imagesFiles">
-                Select or drag &amp; drop images where {{gcp.name}} is visible, then click it in each image
+                Select or drag &amp; drop images where {{gcp.name}} is visible, then click it in each image or use the aumatic detection button.
```
Comment (pierotofy) [inline src/app/images-tagger/images-tagger.component.html:7]: *automatic

**Comment by pierotofy** [comment]
Like the GCP autodetect button. A few improvements are needed:

 * It's not clear to the user that images are being sorted by distance to the GCP. *There needs to be a button that toggles on/off the ability to display images that are outside of a certain radius*. The workflow should be: I drag & drop images, the ones that are outside of a (user specified) radius are not displayed (you can save tons of DOM rendering time). *The idea is that I should be able to drag & drop thousands of images and display only the few that are within a radius of the GCP*. Sorting them does help me distinguish where one GCP ends and another one begins! (The browser also crashes if I drop thousands of images).
 * I should not need to press "Auto Detect" on each image. A global button should try to detect all GCPs for all images I've dropped.
 * A text status along with the spinning gear icon would help end users understand what's going on (e.g. "Reading image %(num) of %(total)", "Detecting GCP in %(filename)", etc.)

**Comment by HeDo88TH** [comment]
Done @pierotofy 🚀

**Comment by HeDo88TH** [comment]
Everything fixed, plus I added pagination to prevent crashes 🚀

**Pull request merged**